pkg/metrics/controller_test.go

```diff
@@ -15,21 +15,94 @@
 package metrics
 
 import (
+	"context"
+	"fmt"
+	"reflect"
 	"strings"
 	"testing"
 
+	"go.opencensus.io/metric/metricdata"
+	"go.opencensus.io/metric/metricexport"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
 	"agones.dev/agones/pkg/apis/stable/v1alpha1"
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/pkg/errors"
 	"github.com/stretchr/testify/assert"
-	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-func TestControllerGameServerCount(t *testing.T) {
+type metricExporter struct {
+	metrics []*metricdata.Metric
+}
 
-	registry := prometheus.NewRegistry()
-	_, err := RegisterPrometheusExporter(registry)
-	assert.Nil(t, err)
+func (e *metricExporter) ExportMetrics(ctx context.Context, metrics []*metricdata.Metric) error {
+	e.metrics = metrics
+	return nil
+}
+
+func serialize(args []string) string {
+	return strings.Join(args, "|")
+}
+
+type expectedMetricData struct {
+	labels []string
+	val    interface{}
+}
+
+func verifyMetricData(exporter *metricExporter, metricName string, expected []expectedMetricData) error {
```

Comment:
this could be:
```
func assertMetricData(t *test.T, exporter *metricExporter, metricName string, expected []expectedMetricData) bool
```

And then you could use the `assert` library all the way through, and wouldn't need to use reflect, and if there is a difference between deep structs, you will get nicer reporting out of the test. WDYT?